tensorboard/plugins/pr_curve/tf_pr_curve_dashboard/tf-pr-curve-card.html

```diff
@@ -0,0 +1,411 @@
+                ChartHelpers.Y_TOOLTIP_FORMATTER_PRECISION);
+            const formatValueOrNaN =
+                (x) => isNaN(x) ? 'NaN' : valueFormatter(x);
+            return [
+              {
+                title: 'Name',
+                evaluate: (d) => d.dataset.metadata().name,
+              },
+              {
+                title: 'Precision',
+                evaluate: (d) => formatValueOrNaN(d.datum.precision),
+              },
+              {
+                title: 'Recall',
+                evaluate: (d) => formatValueOrNaN(d.datum.recall),
+              }];
+          },
+        }
+      },
+      observers: [
+        'reload(runs, tag)',
+        '_runsChanged(_attached, runs)',
+        '_setChartData(_runToPrCurveEntry, _previousRunToPrCurveEntry, _setOfRelevantRuns)',
+        '_updateRunToPrCurveEntry(_runToDataOverTime, runToStepCap)',
+      ],
+      _computeRunColor(run) {
+        return this._colorScaleFunction.scale(run);
+      },
+      attached() {
+        // Defer reloading until after we're attached, because that ensures that
+        // the requestManager has been set from above. (Polymer is tricky
+        // sometimes)
+        this._attached = true;
+        this.reload();
+      },
+      reload() {
+        if (!this._attached) {
+          return;
+        }
+        if (this.runs.length === 0) {
+          // There are no selected runs.
+          this.set('_runToDataOverTime', {});
+          return;
+        }
+
+        this._canceller.cancelAll();
+        const router = getRouter();
+
+        let url = router.pluginRoute('pr_curves', '/pr_curves');
+        url += url.indexOf('?') > -1 ? '&' : '?';
+        url += `tag=${this.tag}`;
+        url += this.runs.map(r => `&run=${r}`).join('');
+
+        const updateData = this._canceller.cancellable(result => {
+          if (result.cancelled) {
+            return;
+          }
+
+          const runToDataOverTime = result.value;
+          this.set('_runToDataOverTime', runToDataOverTime);
+          this.set('loading', false);
+        });
+        this.set('loading', true);
+        this.requestManager.request(url).then(updateData);
+      },
+      _setChartData(
+          runToPrCurveEntry, previousRunToPrCurveEntry, setOfRelevantRuns) {
+        _.forOwn(runToPrCurveEntry, (entry, run) => {
+          const previousEntry = previousRunToPrCurveEntry[run];
+          if (previousEntry &&
+              runToPrCurveEntry[run].step === previousEntry.step) {
+            // The PR curve for this run does not need to be updated.
+            return;
+          }
+
+          if (!setOfRelevantRuns[run]) {
+            // Clear this dataset - the user has unselected it.
+            this._clearSeriesData(run);
+            return;
+          }
+
+          this._updateSeriesDataForRun(run, entry);
+        });
+      },
+      _updateSeriesDataForRun(run, entryFor1Step) {
```

Comment:
`s/1/One`, please. But perhaps `prCurveEntry` is clearer anyway—your choice.

Reply:
Hmm. Yeah, maybe `entryForOneStep` for now.